### PR TITLE
Added unit test Earthworks_01

### PR DESCRIPTION
**Pull request by blaz-p**
# IFC4.3 Unit test

## Unit test: Earthworks_01
This unit test contains an embankment of infrastructure facility using the ifcEarthworksFill concept.

## Review requested
- [IfcEarthworksFill](https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC2/HTML/link/ifcearthworksfill.htm)
- [IfcEarthworksFillTypeEnum](https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC2/HTML/link/ifcearthworksfilltypeenum.htm)
- [IfcTriangulatedIrregularNetwork](https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC2/HTML/link/ifctriangulatedirregularnetwork.htm)

@pjanck

### CODE REVIEW

File: IFC 4.3/Earthworks_01/UT_Earthworks_01.ifc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#38= IFCGEOGRAPHICELEMENT('071iuxsZ189wc6zewS91Ly',#2,$,$,$,#25,#39,$,.TERRAIN.);
```
Comment (pjanck) [inline IFC 4.3/Earthworks_01/UT_Earthworks_01.ifc:46]: In IFC4x3_RC2, `IfcGeographicElement` is deprecated (not noted in documentation, but in its TypeEnum). Use `IfcGeoModel` with `IfcSolidStratum`. Representation with TIN can stay.

File: IFC 4.3/Earthworks_01/readme.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+- IfcAxis2Placement3D
+- IfcCartesianPoint
+- IfcDirection
+- IfcUnitAssignment
+- IfcSIUnit
+- IfcLocalPlacement
+- IfcRelAggregates
+- IfcProductDefinitionShape
+- IfcShapeRepresentation
+- IfcRelContainedInSpatialStructure
```
Comment (pjanck) [inline IFC 4.3/Earthworks_01/readme.md:40]: No need to state these, as they are covered by ProjectSetup_1 and Tin_1.

File: IFC 4.3/Earthworks_01/readme.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+### Content
+
+This scenario covers the following concepts and/or IFC entities:
+
+- IfcEarthworksFill
+- IfcGeographicElement
+- IfcTriangulatedIrregularNetwork
+- IfcCartesianPointList3D
+- IfcEarthworksFillTypeEnum
```
Comment (pjanck) [inline IFC 4.3/Earthworks_01/readme.md:22]: Is deprecated.

**Review by pjanck — CHANGES_REQUESTED** [review]
I propose to cojoin all four EarthworksFill's products and geometries into one. TIN as geometry can stay.

File: IFC 4.3/Earthworks_01/UT_Earthworks_01.ifc (hunk 1 above)
Comment (AlexBrad1ey) [reply IFC 4.3/Earthworks_01/UT_Earthworks_01.ifc:46]: CORRECTION @pjanck 
IfcGeoModel is a geotechnical Assembly and is in essence the same as IfcElementAssembly in it is a aggregator, if the terrain is a singular surface model it should be IfcSolidStratum, if there are multiple layers defined then each layer is a IfcXStratum and grouped into the IfcGeoModel.

Comment (pjanck) [reply IFC 4.3/Earthworks_01/UT_Earthworks_01.ifc:46]: Thank you @AlexBrad1ey for the correction! I agree wholehaeartedly.

**Comment by AlexBrad1ey** [comment]
### Advisory

- there are **2 `IfcSites`** is this intensional, IMO multiple sites should only be used for specific use cases or a project working over multiple disjointed areas.

- Your top level facility is an `IfcBuilding` buy the look of the Model this is not really appropriate I would suggest Road or Rail facility or just a generic IfcFacility (dont forget no PredefinedType requires ObjectType to be set)

Other than that this looks fine to me.

File: IFC 4.3/Earthworks_01/readme.md (hunk 2 above)
Comment (blaz-p) [reply IFC 4.3/Earthworks_01/readme.md]: Deleted in a new commit.

File: IFC 4.3/Earthworks_01/readme.md (hunk 3 above)
Comment (blaz-p) [reply IFC 4.3/Earthworks_01/readme.md:18]: IfcGeographicElement was replaced with IfcSolidStratum.

File: IFC 4.3/Earthworks_01/UT_Earthworks_01.ifc (hunk 1 above)
Comment (blaz-p) [reply IFC 4.3/Earthworks_01/UT_Earthworks_01.ifc]: I replaced IfcGeographicElement with IfcSolidStratum, but I don't know where/how to use IfcElementAssembly?

**Comment by blaz-p** [comment]
Thank you @AlexBrad1ey for your feedback.
Two **IfcSite** were there by mistake. I removed one and replaced **IfcBuilding** with **IfcFacility**.

**Comment by blaz-p** [comment]
> ### Advisory
> * there are **2 `IfcSites`** is this intensional, IMO multiple sites should only be used for specific use cases or a project working over multiple disjointed areas.
> * Your top level facility is an `IfcBuilding` buy the look of the Model this is not really appropriate I would suggest Road or Rail facility or just a generic IfcFacility (dont forget no PredefinedType requires ObjectType to be set)
> 
> Other than that this looks fine to me.

**Comment by pjanck** [comment]
We're moving to the new repository: https://github.com/bSI-InfraRoom/IFC-infra-unit-test

Please open a new pull request there and close this one.

**Comment by blaz-p** [comment]
New pull request was opened in the new repository: https://github.com/bSI-InfraRoom/IFC-infra-unit-test/pull/37.

**Pull request closed without merging**